v2/gcs-to-sourcedb/src/main/java/com/google/cloud/teleport/v2/templates/dao/BaseDao.java

```diff
@@ -88,15 +76,37 @@ public void batchWrite(List<String> batchStatements) throws SQLException {
         }
         statement.executeBatch();
         status = true;
+      } catch (org.postgresql.util.PSQLException e) {
```

Comment:
Can the Postgres and MySQL specific handing be done in the respective Dao? This is BaseDao and we are handling specific exceptions.
Maybe this method can throw the exception and the MySQLDao and PostgreSQLDao can catch the specific exceptions.

Reply:
This has been handled by using the sqlState parameter and checking on specific states.

The connection SQL exception states has been handled as per the error codes mentioned [here].(https://www.postgresql.org/docs/current/errcodes-appendix.html#:~:text=Class%2008%20%E2%80%94%20Connection%20Exception).

Reply:
Specific exceptions for MySQL and PostgreSQL has been refactored and moved to their respective DAOs.